prefs.js

```diff
@@ -184,17 +186,79 @@ function accelTab(notebook) {
 
     notebook.append_page(...makePage('Shortcuts', accelGrid));
 }
+function getActiveLayouts() {
+    const settings = getSettings('layouts');
+    return Object.keys(layouts).filter(entry => {
+        return settings.get_boolean(entry.toString()) === true;
+    });
+}
+function getDefaultLayoutModel() {
+    let model = new Gtk.ListStore();
+    model.set_column_types([GObject.TYPE_STRING, GObject.TYPE_STRING]);
+    let activeLayouts = getActiveLayouts();
+    activeLayouts.forEach(layout => {
+        model.set(model.append(), [0, 1], [layout, layout]);
+    });
+    return model;
+}
+function registerDefaultLayoutBoxListener(model) {
+    const settings = getSettings('layouts');
+    defaultLayoutComboBox.connect('changed', (entry) => {
+        let [success, iter] = defaultLayoutComboBox.get_active_iter();
+        if (!success) return;
+        let value = model.get_value(iter, 0);
+        settings.set_string('defaultlayout', value);
+    });
+}
+function getDefaultLayoutCheckbox() {
+    const settings = getSettings('layouts');
+    let model = getDefaultLayoutModel();
+    let activeLayouts = getActiveLayouts();
+    defaultLayoutComboBox = new Gtk.ComboBox({ model: model });
+    let renderer = new Gtk.CellRendererText();
+    defaultLayoutComboBox.pack_start(renderer, true);
+    defaultLayoutComboBox.add_attribute(renderer, 'text', 1);
+    let defaultLayout = settings.get_string('defaultlayout');
+    defaultLayoutComboBox.set_active(activeLayouts.indexOf(defaultLayout));
+    registerDefaultLayoutBoxListener(model);
+    return defaultLayoutComboBox;
+
+
+}
 
 function layoutsTab(notebook) {
     const settings = getSettings('layouts');
+    // update the list of active layouts every time a layout-switch is toggled
+    for (const [layoutKey, layoutDescription] of Object.entries(layouts)) {
+        settings.connect(`changed::${layoutKey}`, () => {
+            // update the model of the combobox
+            let model = getDefaultLayoutModel();
+            defaultLayoutComboBox.set_model(model);
+            let activeLayouts = getActiveLayouts();
+            let defaultLayout = settings.get_string('defaultlayout');
+            defaultLayoutComboBox.set_active(activeLayouts.indexOf(defaultLayout));
+            registerDefaultLayoutBoxListener(model);
+        });
+    }
+    let setDefaultLayoutAdded = false;
     const layoutItemCreator = (rows, [layout, description]) => {
+        if (!setDefaultLayoutAdded) {
+            let cbox = getDefaultLayoutCheckbox();
+            rows.push(
+                makeItemRow(
+                    'Default layout',
+                    'Determines the default layout for Material Shell. This will only affect the workspaces that will be created after the selection',
+                    cbox
+                )
+            );
+            setDefaultLayoutAdded = true;
+        }
```

Comment:
Instead of having this boolean I would prefer that you create the row outside of the layoutItemCreator Functions that is meant to build the layouts toggles and then add it to the rows at the end like this:
```
let defaultLayoutRow = makeItemRow(...)

const layoutItemCreator = (...)

notebook.append_page(
        ...makePage(
            'Layouts',
            makeItemList([defaultLayoutRow, ...Object.entries(layouts).reduce(layoutItemCreator, [])])
```

Reply:
done